### ISSUE TITLE
"Permission denied" when running "tmt run provision" from a non-root user in a container

### ISSUE THREAD

**Issue by ukulekek** [teemtee/tmt#1065]
Hello, tmt fails to run from a non-root user in a container because it can't create a directory `/run/user/<UID>` for ssh socket.  

Reproducer:
```
$ docker pull quay.io/testing-farm/tmt-all
$ docker run --rm -it --entrypoint bash quay.io/testing-farm/tmt-all
    # Inside the container:
# useradd testuser
# su testuser
$ tmt run provision --how connect --guest 127.0.0.1   # IP doesn't really matter here
/var/tmp/tmt/run-002

/plans/features/advanced
    provision
        how: connect
        guest: 127.0.0.1
        user: root
        key: not provided
Traceback (most recent call last):
  File "/usr/bin/tmt", line 8, in <module>
    tmt.cli.main()
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1700, in invoke
    return _process_result(rv)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1637, in _process_result
    value = ctx.invoke(self._result_callback, value, **ctx.params)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib/python3.10/site-packages/tmt/cli.py", line 337, in finito
    click_context.obj.run.go()
  File "/usr/lib/python3.10/site-packages/tmt/base.py", line 1619, in go
    plan.go()
  File "/usr/lib/python3.10/site-packages/tmt/base.py", line 924, in go
    step.go()
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 110, in go
    plugin.guest().details()
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 365, in details
    distro = self.execute('cat /etc/os-release')[0].strip()
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 479, in execute
    self._ssh_command() + interactive + [self._ssh_guest()] +
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 298, in _ssh_command
    self._ssh_master_connection(command)
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 281, in _ssh_master_connection
    command = command + self._ssh_options() + ["-MNnT", self._ssh_guest()]
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 273, in _ssh_options
    options.extend(["-S", self._ssh_socket()])
  File "/usr/lib/python3.10/site-packages/tmt/steps/provision/__init__.py", line 249, in _ssh_socket
    os.makedirs(socket_dir, exist_ok=True)
  File "/usr/lib64/python3.10/os.py", line 215, in makedirs
    makedirs(head, exist_ok=exist_ok)
  File "/usr/lib64/python3.10/os.py", line 225, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/run/user/1000'
```  

It is reproducible inside every container that doesn't have `/run/user/<UID>` directory.  

Version:
tmt version: 1.10.0 (864b392)  

I'm not sure if it is a bug or expected behavior. Should every container have the directory to make tmt work or is there a possibility to make tmt know how to handle this case?

**Comment by RHEmployee** [comment]
Hello. Would you please test with:
https://github.com/teemtee/tmt/pull/1119

it uses /tmp/ in case where /run/user/<UID> is not provided by systemd (aka: user is not logged in).

**Comment by psss** [comment]
Yes, this should be fixed in `tmt-1.12`. Could you please try the latest version from [copr](https://copr.fedorainfracloud.org/coprs/psss/tmt/) or Fedora updates testing repo?

**Comment by ukulekek** [comment]
Sorry for the late response. I can confirm it works with `tft-1.13`. Thank you! Closing the issue
